content/getting-started/connect-the-pim-4x/step-2.md

```diff
@@ -0,0 +1,120 @@
+_For our example, you will need to create and update products in the PIM so we will mainly need this endpoint: [Patch](/api-reference.html#patch_products) on products._
+
+### 5- Think about your architecture
+Choose the architecture you want for your connector. Indeed, you will have two options here:
+    - You have access to the third-party code source and can develop your connector directly in it.
+    - You can otherwise create a middleware that will sit right in the middle of the tird-party you want to connect and the PIM.
+
+#### In this tutorial
+_For our example, we choose the second option there is little chance that your ERP allows developing into its base code. So you will have to develop your connector as a middleware._
+
+### 6- Ochestrate your global connector
+
+A crucial step. In most cases, you will need to manually map some PIM information with the data sent or received in the software you want to connect.
+
+#### In this tutorial
+_For our example, here is the global orchestration of our connector. First, every evening, you will extract the list of products that were created/updated during the day, in an XML file. Then, this file is read by your connector. ERP products are transformed into PIM products, by applying a mapping of the ERP information with the right attributes of the PIM. To finish, the connector sends the PIM products by using the [patch](/api-reference.html#patch_products) endpoint of our API._
+
+## Some pro tips
+
+As we want you to build the best connector ever, here are some tips to help you cook it the right way.
+
+### Start small
+The use case we took here is really simple, as you can see. In real life, connections maybe a bit more tricky with a lot of different use cases to handle.
+
+We highly recommend you to **start small**. Focus on one single use case at first. Then, when the first one is robust and approved, add some more.
+
+Don't try to do everything at once, they are high chances you won't see the end of it and your connector won't be usable.
+
+### UI or not UI?
+
+Some connectors also require the developement of a UI. For example, it can be very useful in the case you want your final users to be able to input some configuration. Indeed, in some cases, you may want to ask them for product, channel or locale selections.
+
+Another option is to provide a configuration file to input this configuration. It's usually faster to develop but keep in mind that you will be far from delivering the best UX for your final users. :wink:
+
+### Think scalability and performance
+
+It is really important that you take into consideration the scalability and performance of your connector.
+
+As stated previously, again, we advise you to start small. Benchmark your connector with a small volume, then test it with a real-life catalog volume.
+
+To help you in this task, here are examples of questions you can ask yourself:
+- _"How many products, assets, reference entities or options your connector will handle?"_
```

Comment:
I would link each entity to its definition in the documentation, as your readers may not know what reference entities are.

Reply:
```suggestion
- _"How many [products](/documentation/resources.html#product), [product models](/documentation/resources.html#product-model-2x-only), [assets](/documentation/resources.html#asset-since-21-and-ee-only-deprecated), [reference entity records](/documentation/resources.html#reference-entity-record-3x-and-ee-only) or [options](/documentation/resources.html#attribute-option) your connector will handle?"_
```